File: supabase/migrations/run_migrations.js
const fs = require('fs');
const { Client } = require('pg');
require('dotenv').config({ path: '../../.env' });

async function runMigrations() {
  const dbUrl = process.env.DATABASE_URL;
  if (!dbUrl) {
    console.error("DATABASE_URL must be set in .env");
    process.exit(1);
  }

  const client = new Client({
    connectionString: dbUrl,
  });

  try {
    await client.connect();
    console.log("Connected to database.");

    const files = [
      '001_initial_schema.sql',
      '002_auth_schema.sql'
    ];

    for (const file of files) {
      const sql = fs.readFileSync(file, 'utf8');
      console.log(`Applying ${file}...`);
      await client.query(sql);
      console.log(`Successfully applied ${file}`);
    }

    console.log("All migrations applied successfully!");
  } catch (err) {
    console.error("Migration failed:", err);
    process.exit(1);
  } finally {
    await client.end();
  }
}

runMigrations();

File: supabase/migrations/run_migrations.go
package main

import (
	"fmt"
	"os"
	"os/exec"
)

func main() {
	cmd := exec.Command("node", "run_migrations.js")
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr

	if err := cmd.Run(); err != nil {
		fmt.Fprintf(os.Stderr, "migration runner failed: %v\n", err)
		os.Exit(1)
	}
}
